### PR TITLE
feat(project.onboarding) : disable discovery project creation for US region

### PR DESCRIPTION
**Pull request by brofik**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/pci-discovery` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #TAPC-515 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

**Review by y4nnL — APPROVED** [review]

File: packages/manager/apps/public-cloud/src/index.routes.js [hunk 1]
```diff
@@ -22,8 +22,10 @@ export default /* @ngInject */ ($stateProvider, $urlRouterProvider) => {
   $stateProvider.state('app.redirect', {
     url: '?onboarding',
     resolve: {
-      redirect: /* @ngInject */ ($state) =>
-        $state.go('pci.projects.onboarding'),
+      redirect: /* @ngInject */ ($state, coreConfig) =>
+        coreConfig.getRegion() === 'US'
+          ? $state.go('pci.projects.onboardingUs')
+          : $state.go('pci.projects.onboarding'),
```
Comment (ghyenne) [inline packages/manager/apps/public-cloud/src/index.routes.js:28]: I think you use the method redirectTo: (transition) => {

File: packages/manager/modules/pci/src/projects/onboarding-us/onboarding-us.controller.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import illustration from './onboarding-us.png';
+
+export default class {
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/onboarding-us/onboarding-us.controller.js:3]:
```suggestion
export default class PciProjectsOnboardingUsCtrl{
```

File: packages/manager/apps/public-cloud/src/index.routes.js (hunk 1 above)
Comment (brofik) [reply packages/manager/apps/public-cloud/src/index.routes.js:28]: Is there a reason to refactor the `redirect` function of the `plublic-cloud` router ?

**Review by anooparveti — CHANGES_REQUESTED** [review]
Instead of changing the state name for a region. You can handle it internally. Refer https://github.com/ovh/manager/pull/10179/files#diff-9b4d1c1eb591f8bf1ed0b33dea92effcc0ee5ab95abc7fad4faf8128a7528f46
I think this is a cleaner approach as it does not require a duplication of state.

File: packages/manager/modules/pci/src/projects/onboarding/onboarding.routing.js [hunk 3]
```diff
@@ -11,53 +11,56 @@ export default /* @ngInject */ ($stateProvider) => {
...
-          const currentDate = moment().format('YYYY-MM-DD');
-          return {
-            agreements: false,
-            description: `Project ${currentDate}`,
-            cart,
-            hds: cart?.hdsItem !== undefined,
-          };
-        }),
+      isUsRegion: /* @ngInject */ (coreConfig) =>
+        coreConfig.getRegion() === 'US',
```
Comment (darsene) [inline packages/manager/modules/pci/src/projects/onboarding/onboarding.routing.js:37]: No need to enable it with feature flipping ?

**Comment by brofik** [comment]
> Instead of changing the state name for a region. You can handle it internally. Refer https://github.com/ovh/manager/pull/10179/files#diff-9b4d1c1eb591f8bf1ed0b33dea92effcc0ee5ab95abc7fad4faf8128a7528f46 I think this is a cleaner approach as it does not require a duplication of state.

done

File: packages/manager/modules/pci/src/projects/onboarding/onboarding.routing.js (hunk 3 above)
Comment (brofik) [reply packages/manager/modules/pci/src/projects/onboarding/onboarding.routing.js:37]: Yes

**Review by darsene — APPROVED** [review]

**Pull request merged**